src/App.js

```diff
@@ -1,39 +1,50 @@
 import React from 'react';
 import axios from 'axios';
-import { History, NavBar } from './Components';
-import { ReactComponent as MoreIcon } from './assets/MoreIcon.svg';
-import { ReactComponent as Likes } from './assets/Likes.svg';
-import { ReactComponent as Star } from './assets/Star.svg';
-import './App.css';
-import {
-	AppContainer,
-	HistorysContainer,
-	PostContainer,
-	PostsContainer,
-	More,
-	HeaderStatus,
-	GlobalStyle,
-} from './App-styled';
+import { Switch, Route } from 'react-router-dom';
+import { NavBar } from './Components';
+import { PostPage, SearchCollection } from './Pages';
+import { AppContainer, GlobalStyle } from './App-styled';
 
 export default class App extends React.Component {
 	state = {
 		photos: [],
 		isLoading: false,
 		hasError: false,
+		counts: 5,
 	};
 	getPhoto = async () => {
 		try {
 			this.setState({ isLoading: true });
 			const { data } = await axios.get(
-				`https://api.unsplash.com/photos/random?client_id=${process.env.REACT_APP_ACCESS_KEY}&count=5`
+				`https://api.unsplash.com/photos/random?client_id=${process.env.REACT_APP_ACCESS_KEY}&count=${this.state.counts}`
 			);
-			this.setState({ photos: [...this.state.photos, data] });
+			this.setState({
+				photos: [...this.state.photos, ...data],
+				counts: 0,
+				isLoading: false,
+			});
 		} catch (error) {
 			this.setState({ isLoading: false, hasError: true });
 			console.error(error);
 		}
 	};
+	infiniteScroll = () => {
+		// End of the document reached?
+		if (
+			Math.round(window.innerHeight + document.documentElement.scrollTop) ===
+			document.documentElement.offsetHeight
+		) {
+			let newPage = this.state.counts;
+			newPage++;
+			this.setState({
+				counts: newPage,
+			});
+			this.getPhoto();
+		}
+	};
+	window;
```

Comment:
use this npm package for infinite scroll https://www.npmjs.com/package/react-infinite-scroll-component